### ISSUE TITLE
/dispatch-engage-oncall failed with the error "ssl_cacert"

### ISSUE THREAD

**Issue by mccannm1** [Netflix/dispatch#1539]
Hello
I have dispatch integration with Slack setup using event mode and the event URL can reach the 
Request URL according to [Configuring Slack](https://hawkins.gitbook.io/dispatch/administration-guide/incident/plugins/configuring-slack#slack-configuration)

I have a Slack channel which has my 'Netflix Dispatch' integrated with and dispatch commands are available.
When I issue this command /dispatch-engage-oncall   
  I get failed with the error "ssl_cacert".
  
Q. Is this because I need to place an NGINX webserver in docker in front of the dispatch web container, to configure 
 an SSL connection. It seems like it is looking for an SSL cert somewhere but cannot find one.

[How to NGINX Reverse Proxy with Docker Compose](https://dzone.com/articles/how-to-nginx-reverse-proxy-with-docker-compose)

Any help greatly appreciated , thank you
/Martin

**Comment by kevgliss** [comment]
Yes, slack will only send events to endpoints that have a valid SSL certificate. Depending on how you want to deploy Dispatch you have a few options:

For development:
- Use something like https://ngrok.com/ to set up a secure reverse proxy.

For production:
- Deploy something to terminate and serve an SSL certificate (either NGNIX, apache, or something like an AWS ELB). 

For production or development:
- Use 'socket mode' (see: https://hawkins.gitbook.io/dispatch/administration-guide/incident/plugins/configuring-slack#socket-mode)

Socket mode does not require a publically available endpoint, which is usually easier for a lot of organizations to deploy. The only caveat with 'socket mode' is it will require you to run an additional process to set up and monitor the slack WebSockets connection.
